src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -128,25 +127,8 @@ public NodeStateTable<T> AsCached()
             return (_states[^1].GetItem(0), HasTrackedSteps ? Steps[^1] : null);
         }
 
-        public ImmutableArray<NodeStateEntry<T>> Batch()
-        {
-            var sourceBuilder = ArrayBuilder<NodeStateEntry<T>>.GetInstance();
-            foreach (var entry in this)
-            {
-                // If we have tracked steps, then we need to report removed entries to ensure all steps are in the graph.
-                // Otherwise, we can just return non-removed entries to build the next value.
-                if (entry.State != EntryState.Removed || HasTrackedSteps)
-                {
-                    sourceBuilder.Add(entry);
-                }
```

Comment:
@cston  the `HasTrackedSteps` was to make sure that we retruend this entry from the iterator so we could associated it with the perf data we're collecting.  On the receiver side though we then filtered this out from the actual entry-list we were creating, but we used it for the step-list being built.